### DISCUSSION TITLE
GiveWP Webhooks and/or Triggers

### DISCUSSION THREAD

**Discussion by malucy2022** [impress-org/givewp#6602] [Developer Q&A]
Our GiveWP sales rep mentioned to us that there are webhooks and/or triggers natively available in Give. We have not been able to find any documentation. I scoured as best I could the Give front-end and was not able to find webhooks. 

We are using Zapier right now however with our volume and giving Tuesday coming up, we need to move off of Zapier and onto our own server (code our own solution).

I have also searched here the term webhook and trigger and not found anything. Essentially, we want to fire an event from WP for specific events (i.e. New Donation, New Donor, etc.) and send the GiveWP payload to our 3rd party system. 

Can someone help point me in the direction of where Webhooks and/or Triggers are within GiveWP.

Thanks ahead of time
~Mike

**Comment by JoshuaHungDinh** [comment]
Hi @malucy2022! We are excited to hear that you are using GiveWP’s platform to assist with your donations. At the moment we are in the process of exciting new features that will transform how GiveWP integrates with payments and events. These changes will improve consistency and create a more user friendly environment for developers to work with. Currently there is not much in terms of documentation for triggers as the new system is still going through changes.  Some hooks that you may find useful include ‘give_insert_payment’,   'givewp_donor_created', 'givewp_donor_updated’,  'givewp_donation_updated’ and ‘givewp_donation_creating’. We appreciate your patience as we work toward making GiveWP a better place for it's users. Happy holidays!

**Comment by Rsmith-gs** [comment]
Hello @JoshuaHungDinh It's been a while since your post and with the launch of GiveWP 3.0, as of posting this the current version is 3.6.2. Are the hooks you listed still all active? Are there any new ones? Is there any new documentation available?

> ‘give_insert_payment’, 'givewp_donor_created', 'givewp_donor_updated’, 'givewp_donation_updated’ and ‘givewp_donation_creating’

**Reply by JoshuaHungDinh** [reply]
Hello @Rsmith-gs, as of the current version 3.6.2 of GiveWP, the hooks you listed: give_insert_payment, givewp_donor_created, givewp_donor_updated, givewp_donation_updated, and givewp_donation_creating are all still active and available for use. I understand you are looking for new hooks. Is there something specific you are trying to accomplish?

As for documentation, it is still in progress but we are aiming to provide it to users sometime in the near future. You can keep an eye on the official GiveWP documentation page for any new additions or updates. In the absence of specific documentation for newer versions, exploring the plugin's source code might provide some insight.

**Reply by Rsmith-gs** [reply]
Hello @JoshuaHungDinh, I was suggested to look at the discussions by Alex Coleman regarding a ticket I had regarding utilizing the current [MailChimp add-on](https://givewp.com/documentation/add-ons/mailchimp/) to send data to [MailPoet](https://www.mailpoet.com/) instead. I've got a proof of concept written up, but was wondering if you could confirm that it will do what I want and potentially look into making a dedicated MailPoet add-on for those who use MailPoet instead.

Thanks for your help!
